components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/AESDecryptor.java

```diff
@@ -0,0 +1,33 @@
+package org.wso2.carbon.identity.sso.agent;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.apache.commons.codec.binary.Base64;
+
+public class AESDecryptor {
+
+    private static final String ALGORITHM =  "AES/CBC/PKCS5Padding";
+
+    public static String decode(String encryptedSecret,
+                                String cipherKey)
+            throws Exception {
```

Comment:
I think this can be taken to a single line? Please check with the 120 length limit.